apps/ledger-live-desktop/tests/utils/xray.service.ts

```diff
@@ -0,0 +1,81 @@
+import axios from "axios";
+import fs from "fs";
+
+class XrayService {
+  private static baseUrl = "https://xray.cloud.getxray.app/api/v2";
+  private static token: string;
+
+  private static async authenticate(): Promise<string> {
+    const clientId = process.env.XRAY_CLIENT_ID;
+    const clientSecret = process.env.XRAY_CLIENT_SECRET;
+
+    if (!clientId || !clientSecret) {
+      throw new Error("Missing XRAY_CLIENT_ID or XRAY_CLIENT_SECRET environment variables");
+    }
+
+    const auth: Auth = {
+      client_id: clientId,
+      client_secret: clientSecret,
+    };
+
+    try {
+      const response = await axios.post(`${this.baseUrl}/authenticate`, auth, {
+        headers: { "Content-Type": "application/json" },
+      });
+      this.token = response.headers["x-access-token"];
+      console.log(`Authentication successful`);
+      return this.token;
+    } catch (error) {
+      console.error(`Error during authentication: ${error}`);
+      throw error;
+    }
+  }
+
+  public static async importExecution(xmlFilePath: string): Promise<ImportRes> {
+    if (!this.token) {
+      await this.authenticate();
+    }
+
+    try {
+      console.log(`Importing execution from XML file: ${xmlFilePath}`);
+      const response = await axios.post(
+        `${this.baseUrl}/import/execution/junit?projectKey=B2CQA&testPlanKey=${process.env.TEST_PLAN}`,
+        fs.readFileSync(xmlFilePath, "utf-8"),
+        {
+          headers: {
+            "Content-Type": "text/xml",
+            Authorization: `Bearer ${this.token}`,
+          },
+        },
+      );
+
+      if (response.status === 200) {
+        const responseData = response.data;
+        console.log("Import successful, response data:", responseData);
+        return {
+          id: responseData.id,
+          key: responseData.key,
+          self: responseData.self,
+        };
+      } else {
+        throw new Error(`Error during XrayService.importExecution(): Status: ${response.status}`);
```

Comment:
would be worth logging other info than `status` in case of an error (error content)

Reply:
maybe [response data](https://github.com/LedgerHQ/ledger-live/pull/7018/files#diff-f61ac1468fcdc3b8363a29442e0049ba714d3679b323876698cd730995238f23R62) should also be logged in [error thrown](https://github.com/LedgerHQ/ledger-live/pull/7018/files#diff-f61ac1468fcdc3b8363a29442e0049ba714d3679b323876698cd730995238f23R64)